client/ayon_deadline/plugins/publish/global/collect_pools.py

```diff
@@ -69,7 +69,15 @@ def process(self, instance):
             instance.data["secondaryPool"] = None
 
     @classmethod
-    def get_attribute_defs(cls):
+    def get_attr_defs_for_instance(cls, create_context, instance):
+        # Filtering of instance, if needed, can be customized
+        if not cls.instance_matches_plugin_families(instance):
+            return []
+
+        # Attributes logic
+        creator_attributes = instance["creator_attributes"]
+
+        visible = creator_attributes.get("farm", False)
```

Comment:
This is suddenly Houdini-specific (other integration may not use `farm` creator attribute as a boolean but may for example use an enum to decide between local render, farm render, export jobs, etc. and thus I wouldn't approach it this way. In this case I'd reverse it and make the Creator behave dynamically instead.

So what the Creator does is set additional instance data on whether something is considered to be "running on the farm" - and if so, that Creator then adds the `farm` family. As soon as all "farm-compatible" creators do that and track whether it should submit to farm or not then all the deadline plug-ins could potentially end up just targeting `farm` family - and by doing so and with https://github.com/ynput/ayon-core/pull/973 that'd also mean they would suddenly not be displaying the attributes as soon as the farm toggle is disabled by the user.

That also automatically means that with that PR merged all these deadline plug-ins wouldn't need custom logic implementations maybe? (except for maybe the live updating to 'families' changing?)

Reply:
It's good to point that out, thank you. 
Whether it'll be a `farm` family/trait or a creator/publish attribute. I'll be fine and wait happily till we come up with a decision. 
Let's keep this PR as it is (because it's for mainly sharing the code and reproducing the same result in the gif above). 
What do you think ? 

> That also automatically means that with that PR merged all these deadline plug-ins wouldn't need custom logic implementations maybe? (except for maybe the live updating to 'families' changing?)

It'd be cool to add these insights to the refactoring deadline issue/epic?

Reply:
I've reverted any changes in `collect_pools.py` since it's removed in https://github.com/ynput/ayon-deadline/pull/49

Reply:
> Let's keep this PR as it is (because it's for mainly sharing the code and reproducing the same result in the gif above).
What do you think ?

I think it is confusing PR, that might some people use in production with customized addon, because the with changes here it looks like it works, but it does not.